uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/FileSystemLockManager.java

```diff
@@ -0,0 +1,32 @@
+package org.uberfire.java.nio.fs.jgit;
+
+import java.io.File;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+public class FileSystemLockManager {
+
+    final Map<String, FileSystemLock> fileSystemsLocks = new ConcurrentHashMap<>();
+
+    private static class LazyHolder {
+
+        static final FileSystemLockManager INSTANCE = new FileSystemLockManager();
```

Comment:
Adriel, this factory should be syncronized in order to avoid race conditions and multiple object creation (fs manager vs jgitconfigucation)

Reply:
for the record, the hashmap that stores the locks is a concurrent hash map